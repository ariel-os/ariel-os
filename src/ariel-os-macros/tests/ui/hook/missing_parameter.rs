#![no_main]

// FAIL: missing parameter
#[ariel_os::hook(usb_builder)]
fn usb_builder() {}
