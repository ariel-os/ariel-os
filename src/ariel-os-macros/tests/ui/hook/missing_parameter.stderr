error[E0061]: this function takes 0 arguments but 1 argument was supplied
 --> tests/ui/hook/missing_parameter.rs:5:4
  |
4 | #[ariel_os::hook(usb_builder)]
  | ------------------------------ unexpected argument of type `&mut ariel_os::reexports::embassy_usb::Builder<'static, UsbDriver>`
5 | fn usb_builder() {}
  |    ^^^^^^^^^^^
  |
note: function defined here
 --> tests/ui/hook/missing_parameter.rs:5:4
  |
5 | fn usb_builder() {}
  |    ^^^^^^^^^^^
